src/Sentry/Internal/Extensions/JsonExtensions.cs

```diff
@@ -285,10 +293,21 @@ public static void Deconstruct(this JsonProperty jsonProperty, out string name,
         public static void WriteDynamic(
             this Utf8JsonWriter writer,
             string propertyName,
-            object? value)
+            object? value,
+            IDiagnosticLogger? logger)
         {
-            writer.WritePropertyName(propertyName);
-            writer.WriteDynamicValue(value);
+            try
+            {
+                writer.WritePropertyName(propertyName);
+                writer.WriteDynamicValue(value, logger);
+            }
+            catch (Exception e)
+            {
+                // The only location in the protocol we allow dynamic objects are Extra and Contexts
+                // In the event of an instance that can't be serialized, we don't want to throw away a whole event
+                // so we'll suppress issues here.
+                logger?.LogError("Failed to serialize object for property {0}", e, propertyName);
```

Comment:
We could use the same approach @SimonCropp took and if `logger is null) { throw`;

Reply:
I went with  `catch Exception ex when (logger != null)`